src/astarte-client/client.ts

```diff
@@ -288,6 +288,13 @@ astarteAPIurl`${config.realmManagementApiUrl}v1/${'realm'}/interfaces/${'interfa
     return response.data;
   }
 
+  async getInterfaces(): Promise<string[]> {
+    const endpointUri = new URL(this.apiConfig.interfaces(this.config));
+    endpointUri.searchParams.set('detailed', 'true');
+    const response = await this.$get(endpointUri.toString());
+    return response.data;
+  }
```

Comment:
Thanks for the input, I tested this on Realm managment version 1.1 and the itself will not fail if I add the extra param `detailed=true`. My idea was to get data either with full deatils or only names and then check on the RegisterDevice page if I got the full data:
``` ts
if (Array.isArray(interfacesData) && interfacesData.every((interfaceDetail) => typeof interfaceDetail === 'object')
```
If the API version is older, I would fetch other needed data with singular API calls. 
It was easier to implement this without `detailed=true` parameter but it would call API too much times.
If this function `getInterfaces` causes confusion maybe we can implement something along the following lines:
```
if realm_managment_version < 1.2 
    interface_names = get_interface_names()
    fetch_major_and_minor_for_each_interface()
else
    get_interfaces()
```

I tested this with Realm managment 1.1 and 1.2-dev and it is working as intended. Please let me know if I should refactor RegisterDevicePage.tsx to first check for API version and then fetch data or just fetch interface names without `detailed=true` param.

Reply:
Correct, we can't afford to call the API too many times. I think we should split the UI here.
if RealmManagement >= 1.2 -> show the new UI with a dropdown
if RealmManagement < 1.2 -> show some legacy UI. The legacy UI can be the same as the old one, or something where you select the name from a dropdown, and the version selector only appears after you selected the interface name (similar to what we do in the trigger editor)